google-cloud-storage/src/test/java/com/google/cloud/storage/RewindableHttpContentPropertyTest.java

```diff
@@ -85,6 +110,30 @@ static Arbitrary<PathScenario> pathScenario() {
                             .as(PathScenario::of)));
   }
 
+  @Provide("ByteBuffersScenario")
+  static Arbitrary<ByteBuffersScenario> byteBuffersScenarioArbitrary() {
+    return Arbitraries.lazyOf(
+            () ->
+                Arbitraries.oneOf(
+                    byteBuffers(1, 10),
+                    byteBuffers(10, 100),
+                    byteBuffers(100, 1_000),
+                    byteBuffers(1_000, 10_000),
+                    byteBuffers(10_000, 100_000),
+                    byteBuffers(100_000, 1_000_000)))
```

Comment:
Pinged you through chat; but what's the rationale for using multiple buffers versus one? A few reasons come to mind (easier to allocate smaller contiguous memory buffers, individual buffers can be passed as-is to transport writes); and wondering if there are others.

Reply:
One big one for us specifically, is from the buffered to unbuffered layer, this enables the ability to reduce (and in some cases completely remove) copies between the `ByteBuffer` provided to our `write` method, and any underlying location it needs to be propagated to.

A codified test case that illustrates this minimized copying can be seen here: https://github.com/googleapis/java-storage/blob/ed05232889c0442f43eacde5dbc80f44f09ebe83/google-cloud-storage/src/test/java/com/google/cloud/storage/DefaultBufferedWritableByteChannelTest.java#L312-L323

In that test flushes will happen every 3 bytes, where we're doing 16 byte writes. In order to flush, we don't need to copy into our buffer, we can simply take a 3 byte slice of the provided `ByteBuffer` and flush that. Then if there is any outstanding bytes they are enqueued. Then the next time `write` is called, in a single call we can flush our buffer and the slice from the new `ByteBuffer`.